library/std/src/io/stdio.rs

```diff
@@ -677,6 +702,8 @@ impl Write for StdoutLock<'_> {
     fn write_all_vectored(&mut self, bufs: &mut [IoSlice<'_>]) -> io::Result<()> {
         self.inner.borrow_mut().write_all_vectored(bufs)
     }
+    // Don't specialize write_fmt because we need to be sure we don't
+    // reentrantly call borrow_mut
```

Comment:
Doesn't this mean that the logic above of buffering `write_fmt`s is unused here? The provided implementation of `write_fmt` will just call `write` many times, which does not buffer anything with `BufferMode::None`.

Reply:
Yes, that's what'd happen in practice. `SwitchWriter` was designed independently of this particular use case; I wanted it to be internally consistent as an arbitrary write adapter.